src/Our.Umbraco.Ditto/Extensions/PublishedContentExtensions.cs

```diff
@@ -330,43 +330,49 @@ public static class PublishedContentExtensions
+                        // Time custom conversions.
+                        using (DisposableTimer.DebugDuration(type, string.Format("Custom TypeConverter ({0}, {1})", content.Id, propertyInfo.Name), "Complete"))
                         {
-                            // Time custom conversions.
-                            using (DisposableTimer.DebugDuration(type, string.Format("Custom TypeConverter ({0}, {1})", content.Id, propertyInfo.Name), "Complete"))
+                            // Get the custom converter from the attribute and attempt to convert.
+                            var toConvert = Type.GetType(converterAttribute.ConverterTypeName);
+                            if (toConvert != null)
                             {
-                                // Get the custom converter from the attribute and attempt to convert.
-                                var toConvert = Type.GetType(converterAttribute.ConverterTypeName);
-                                if (toConvert != null)
+                                var converter = DependencyResolver.Current.GetService(toConvert) as TypeConverter;
+
+                                if (converter != null)
                                 {
-                                    var converter = DependencyResolver.Current.GetService(toConvert) as TypeConverter;
+                                    // Create context to pass to converter implementations.
+                                    // This contains the IPublishedContent and the currently converting property descriptor.
+                                    var descriptor = TypeDescriptor.GetProperties(instance)[propertyInfo.Name];
+                                    var context = new PublishedContentContext(content, descriptor);
 
-                                    if (converter != null)
+                                    Type propertyValueType = null;
+                                    if (propertyValue != null)
                                     {
-                                        // Create context to pass to converter implementations.
-                                        // This contains the IPublishedContent and the currently converting property descriptor.
-                                        var descriptor = TypeDescriptor.GetProperties(instance)[propertyInfo.Name];
-                                        var context = new PublishedContentContext(content, descriptor);
+                                        propertyValueType = propertyValue.GetType();
+                                    }
 
-                                        if (converter.CanConvertFrom(context, propertyValue.GetType()))
-                                        {
-                                            object converted = converter.ConvertFrom(context, culture, propertyValue);
+                                    if (converter.CanConvertFrom(context, propertyValueType))
```

Comment:
All the TypeConverters would have to be amended to allow `null` in the `CanConvertFrom` method.